app/src/main/kotlin/com/wire/android/ui/connection/ConnectionActionButton.kt

```diff
@@ -111,7 +115,11 @@ fun ConnectionActionButton(
                 text = stringResource(R.string.connection_label_ignore),
                 loading = viewModel.actionableState().isPerformingAction,
                 state = WireButtonState.Error,
-                onClick = viewModel::onIgnoreConnectionRequest,
+                onClick = {
+                    viewModel.onIgnoreConnectionRequest {
+                        onConnectionRequestIgnored((viewModel as ConnectionActionButtonViewModelImpl).userName)
```

Comment:
You could make it so that `onSuccess` returns `userName`, like:
```
fun onIgnoreConnectionRequest(onSuccess: (String) -> Unit)
```
then call `onSuccess(userName)`
and thanks to that we wouldn't have to make this casting here, instead just:
```
viewModel.onIgnoreConnectionRequest { userName ->
    onConnectionRequestIgnored(userName) 
}
```